src/app/page/page.component.ts

```diff
@@ -24,7 +24,14 @@ namespace Page {
           this.previousWidth = this.$window.innerWidth;
         });
       });
-      this.$rootScope.$on(CLOSE_MAIN_NAV_EVENT, () => this.isNavOpen = false);
+      let curr_state = window.localStorage.getItem('defaultVerticalNavState');
+      if (curr_state === "show") {
```

Comment:
pls use enum I suggested

Reply:
and regarding where the `VerticalNavState` enum type should be defined, there are a few options. But I think the simplest is just put it in the `page.component.ts` and then from the `general-preferences.service.ts` reference it.